### PR TITLE
Add series 101.2: Development Environment

### PR DESCRIPTION
**Pull request by karnkaul**

### CODE REVIEW

File: content/series/101/2_development_environment/index.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+#### Compiler Warnings and clang-tidy
+
+Since C++ standards are generally very meticulous about backwards compatibility, not-so-great language and library design decisions made in the past cannot easily be rectified through breaking changes, and older programs must not suddenly stop compiling with a new compiler (provided they weren't engaging in unspecified behaviour in the first place). Ergo, there are a few language potholes and traps where the compiler can only warn you - cannot issue an error and stop (unless you configure it to be that way for a specific set of / all warnings), and which is why modern C++ development is greatly aided by turning up compiler warnings and using linters that perform static analysis outside compilation (perhaps even risking some false positive warnings). `clang++` offers a `g++`-like front-end, and accepts all those compiler flags (and many more); at the bare minimum it is highly recommended to set at least the following, for **every single C++ project you author**:
```
Comment (hentai-chan) [inline content/series/101/2_development_environment/index.md:70]: > Ergo, there are a few language potholes and traps where the compiler can only warn you - cannot issue an error and stop (unless you configure it to be that way for a specific set of / all warnings), and which is why modern C++ development is greatly aided by turning up compiler warnings and using linters that perform static analysis outside compilation (perhaps even risking some false positive warnings).

This is a very long sentence. Your text is more accessible for non-native speakers and younger readers if you break it up a little. For example:

_Ergo, there are a few language potholes and traps where the compiler can only issue warnings for you unless you configure the compiler to interpret (all) warnings as errors. Modern C++ development is greatly aided by turning up compiler warnings and using linters that perform static analysis outside compilation, though this practice sometimes risks false positives._

> [...] to set at least the following __options__ for every [...]

File: content/series/101/2_development_environment/index.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+target_compile_options(foo PUBLIC -Wall -Wextra)
+```
+
+`clang-tidy` can be invoked through the command line, though it is also available as a third-party [editor plugin](https://marketplace.visualstudio.com/items?itemName=notskm.clang-tidy) for VSCode (not authored by LLVM). There should be more integrated support for it within a few years.
+
+> _**Recommended:** [CppCon 2017: Kate Gregory "10 Core Guidelines You Need To Start Using Now"](https://www.youtube.com/watch?v=XkDEzfpdcSg)_
+
+### Building and Debugging
+
+The core idea behind debugging is to spin off a command line task: `cmake --build <output_dir>` which can be invoked by some keyboard shortcut via the text editor. Debugging is a bit more involved, though command line debugging is simpler on Linux than Windows; setting up VSCode to use the native debugger (`gdb` or `cppdbg`) is relatively straightforward on both. On VSCode both these aspects are driven by a JSON file (in `./.vscode`) each.
```
Comment (hentai-chan) [inline content/series/101/2_development_environment/index.md:82]: > The core idea behind debugging is to spin off a command line task: `cmake --build <output_dir>` which can be invoked by some keyboard shortcut via the text editor.

Use different punctuation to guide the reading flow:

Better: _The core idea behind debugging is to spin off a command line task (`cmake --build <output_dir>`) which can be invoked by some keyboard shortcut via the text editor._

**Review by hentai-chan — CHANGES_REQUESTED** [review]

File: content/series/101/2_development_environment/index.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+endif()
+```
+
+Check clangd's logs in VSCode's **Output** window to confirm that it isn't reporting any errors.
+
+> _**Note:** CMake, `compile_commands.json`, and clangd don't work well together when using **precompiled headers**; in such cases create a dummy config as a copy of `Debug` but with PCH off, and copy and use the JSON generated there._
+
+#### clang-format
+
+Most C++ programmers spend a considerable amount of time meticulously formatting their precious code, until they make an acquaintance with an autoformatter - of which `clang-format` is the indisputed champion - after which they pretty much never hand-format code again. This is the secret behind the reams of impossibly perfectly formatted and documented library code, for example throughout the standard library headers.
```
[the remark this answers is not in the document]
Comment (karnkaul) [reply content/series/101/2_development_environment/index.md]: Here `reams` means "bunch of printing paper", is just a lame metaphor. 😛

**Review by hentai-chan — APPROVED** [review]
Well done! 💮

**Pull request merged**